certbot/ocsp.py

```diff
@@ -65,33 +87,103 @@ def ocsp_revoked(self, cert_path, chain_path):
         except errors.SubprocessError:
             logger.info("OCSP check failed for %s (are we offline?)", cert_path)
             return False
-
         return _translate_ocsp_query(cert_path, output, err)
 
 
-    def determine_ocsp_server(self, cert_path):
-        """Extract the OCSP server host from a certificate.
+def _check_ocsp_response_signature(response_ocsp, issuer_cert):
+    """Verify an OCSP response signature against certificate issuer"""
+    try:
+        # TODO: (adferrand 2019-11-01) Following line can be replaced with a direct call to
+        #  response_ocsp.signature_hash_algorithm once cryptography 2.5 is released.
+        #  See https://github.com/pyca/cryptography/issues/4680
+        chosen_hash = x509._SIG_OIDS_TO_HASH[response_ocsp.signature_algorithm_oid]  # pylint: disable=protected-access
+    except KeyError:
+        raise UnsupportedAlgorithm(
+            "Signature algorithm OID:{0} not recognized"
+            .format(response_ocsp.signature_algorithm_oid)
+        )
+
+    issuer_cert.public_key().verify(
```

Comment:
Something is going wrong here. When I test this against LE staging (or prod), I get:
```
Invalid signature for OCSP response on /etc/letsencrypt/live/example.com/cert.pem
```
which doesn't happen when using the openssl command line.

Reply:
Indeed. The signature is failing, and I cannot understand why, I am doing as in the documentation from `cryptography`.

@reaperhulk or @alex, would you mind helping me to find what is wrong with my code ?

Reply:
You're using PSS, but I'm pretty sure the signature is PKCS1v15. Is there a reason you went for PSS?

Reply:
Only because it is written as it in the documentation for general RSA signature verification, but no specific walkthrough on OCSP. It is only noted that the OCSPResponse contains the DER encrypted payload used to build the signature, and that it could be used for verifying the signature, but nothing more.

https://cryptography.io/en/latest/x509/ocsp/#cryptography.x509.ocsp.OCSPResponse.tbs_response_bytes

Reply:
It worked ! Indeed PKCS1v15 is used for OCSP.

Really sorry, everything is explained in the relevant test from cryptography project:
https://github.com/pyca/cryptography/blob/master/tests/x509/test_ocsp.py

May I suggest to maybe link, from `tbs_response_bytes` of `OCSPResponse`, to the example provided in https://cryptography.io/en/latest/x509/reference/#cryptography.x509.Certificate.tbs_certificate_bytes, as it is relevant for this situation ?